.github/workflows/test.yml

```diff
@@ -98,17 +112,20 @@ jobs:
           - dev
     steps:
       - uses: actions/checkout@v4
+        if: contains(needs.check_changes.outputs.path_filters, 'docker')
       - name: publish images
         if: contains(needs.check_changes.outputs.path_filters, 'docker')
         working-directory: docker
-        run: ./publish.sh "${{ matrix.image }}" "dev-${{ github.head_ref || 'latest' }}"
+        run: ./publish.sh "${{ matrix.image }}" "${{ needs.set_docker_image_tag.outputs.tag }}"
 
   check_formatting:
-    needs: build_docker_images
+    needs: 
+      - set_docker_image_tag
+      - build_docker_images
```

Comment:
do jobs like this and below need more than the single job `build_docker_images ` because it might be skipped? 

since it all seems linear and looks like only the steps can be skipped, might be nice if all the `needs:` had a single job listed?

Reply:
To access outputs of any job, they have to be in the "needs"

Reply:
What we could do is pass along the outputs though. So outputs of build_docker_images has the outputs of the prior two jobs